File: packages/dotnet/ReadMe/HarJsonTranslationLogics/MaskHelper.cs
using System;
using System.Security.Cryptography;
using System.Text;

public static class MaskHelper
{
    public static string Mask(string data)
    {
        using (SHA512 sha512 = SHA512.Create())
        {
            byte[] hashBytes = sha512.ComputeHash(Encoding.UTF8.GetBytes(data));
            string base64Hash = Convert.ToBase64String(hashBytes);
            string opts = data.Length >= 4 ? data.Substring(data.Length - 4) : data;
            return $"sha512-{base64Hash}?{opts}";
        }
    }
}

File: packages/php/src/HAR/MaskHelper.php
<?php

namespace ReadMe\HAR;

class MaskHelper
{
    public static function mask(string $data): string
    {
        $hashBytes = hash('sha512', $data, true);
        $base64Hash = base64_encode($hashBytes);
        $opts = substr($data, -4);
        return 'sha512-' . $base64Hash . '?' . $opts;
    }
}
